src/Products/PluginIndexes/common/UnIndex.py

```diff
@@ -304,6 +346,69 @@ def _apply_not(self, not_parm, resultset=None):
     def _convert(self, value, default=None):
         return value
 
+    def _catalog_cache_key(self, catalog):
+        cid = catalog.getId()
+        counter = getattr(aq_base(catalog), 'getCounter', None)
```

Comment:
I know it isn't new to this PR, but where are we expecting `getCounter` to come from?  It isn't a method of `ZCatalog` / `Catalog`.